hack/glide-update.sh

```diff
@@ -3,4 +3,17 @@
 glide update
 
 # need to remove the vendor tree from our types to be able to build against the "right" apimachinery and such for now
-glide install --strip-vendor github.com/openshift/origin
\ No newline at end of file
+glide install --strip-vendor github.com/openshift/origin
+
+dep init
+dep prune
+
+# we shouldn't have modified anything
+git diff-index --name-only --diff-filter=M HEAD | xargs -r git checkout -f
+# we need to preserve code-generator and friends
+git diff-index --name-only HEAD | grep -F -e 'k8s.io/code-generator' -e 'k8s.io/gengo' | xargs -r git checkout -f
+
+# now cleanup what's dangling
+git clean -f -d
+# for some weird reason we have to run this twice to remove a folder _vendor_date/
```

Comment:
@deads2k Perhaps it's because there was a `.gitignore` in the subdirectory.
What happens if run `git clean -x -f -d` once instead of two identical commands?